ArtOfIllusion/src/artofillusion/animation/JointEditorDialog.java

```diff
@@ -432,47 +579,57 @@ public void run()
 
     private void dealWithDrag(MouseDraggedEvent ev)
     {
-      if (dragging == -1)
+      if (!(dragging0 || dragging1 || dragging2 || dragging3 || dragging4))
         return;
       Point pos = ev.getPoint();
-      double angle = Math.atan2(pos.x-OFFSET1-SIZE/2, OFFSET1+SIZE/2-pos.y)*180.0/Math.PI;
-      angle = Math.rint(angle);
-      if (lastAngle-angle > 270.0)
-        angle += 360.0;
-      else if (angle-lastAngle > 270.0)
-        angle -= 360.0;
+
+      double mouseAngle  = getMouseAngleDeg(ev.getPoint());      
+      double angle = Math.rint(mouseAngle);
+
+      if (panel.rangeBox.getState())
+      {
+        if (lastAngle-angle > 320)
+          angle += 360.0;
+        else if (angle-lastAngle > 320)
+          angle -= 360.0;
+      }
+
       double val = panel.dof.pos;
       boolean range = panel.rangeBox.getState();
       double minv = (range ? panel.dof.min : panel.min);
       double maxv = (range ? panel.dof.max : panel.max);
       double minComfort = panel.dof.minComfort;
       double maxComfort = panel.dof.maxComfort;
-      if (dragging != 1 && angle < minv)
+      if (!dragging1 && angle < minv)
```

Comment:
And down through here, we would `case ... switch` on the enum.

Reply:
Actully there is a short sections a few lines down from this, where switch could be used. Not really anywhere else.